src/chatgpt-api.ts

```diff
@@ -400,21 +400,29 @@ export class ChatGPTAPI {
           }
         }, [] as string[])
         .join('\n\n')
-
       const nextNumTokensEstimate = await this._getTokenCount(prompt)
-      const isValidPrompt = nextNumTokensEstimate <= maxNumTokens
-
-      if (prompt && !isValidPrompt) {
+      const validatingRequiredMessages =
+        nextMessages.length <= requiredMessagesAmount
+      const promptLimit = validatingRequiredMessages
+        ? this._maxModelTokens - 1
+        : this._maxModelTokens - this._maxResponseTokens
+      const isValidPrompt = nextNumTokensEstimate <= promptLimit
+
+      if (!isValidPrompt && validatingRequiredMessages) {
+        // SystemMessage and user main prompt exceed the limit. Handling the request without one of them has no sense
+        // Intentionally send wrong data to ChatGPT to return exception to user of the same shape
+        messages = nextMessages
+        numTokens = nextNumTokensEstimate
         break
       }
 
-      messages = nextMessages
-      numTokens = nextNumTokensEstimate
-
       if (!isValidPrompt) {
```

Comment:
This condition almost never happens, because we had `if (prompt && !isValidPrompt) { break }` before. I removed that one